### PR TITLE
Add Tags resource and test

### PR DESCRIPTION
**Pull request by joseconsador**
Some code formatting for PSR2.
Added files: Tags.php , TagsTest.php 
The unit tests check if the correct routes are created based on the chaining.

/cc @miogalang @jwswj @mmolina

### Risks
 - The tags could not be set (low)

### CODE REVIEW

**Comment by joseconsador** [comment]
WIP

**Comment by joseconsador** [comment]
:haircut:

File: src/Zendesk/API/Resources/Tags.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+     *              tickets/{id}/tags.json
+     *              topics/{id}/tags.json
+     *              organizations/{id}/tags.json
+     *              users/{id}/tags.json
+     *
+     * @throws \Exception
+     */
+    public function getRoute($name, array $params = array())
+    {
+        if (!in_array($name, array('update', 'find', 'create', 'delete'))) {
```
Comment (jwswj) [inline src/Zendesk/API/Resources/Tags.php]: Can this array be defined as a variable/constant?

File: src/Zendesk/API/Resources/Tags.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        if (empty($lastChained)) {
+            throw new CustomException('The ' . $name . '() method needs to be called while chaining.');
+        }
+
+        $chainedResourceNames = array_keys($lastChained);
+
+        $id = reset($lastChained);
+        $resource = $chainedResourceNames[0]::OBJ_NAME_PLURAL;
+
+        return "$resource/$id/tags.json";
```
Comment (jwswj) [inline src/Zendesk/API/Resources/Tags.php]: could tags also use `OBJ_NAME_PLURAL`?

File: src/Zendesk/API/Resources/Tickets.php [hunk 3]
```diff
@@ -60,6 +64,7 @@ public function __construct(\Zendesk\API\HttpClient $client)
     {
         parent::__construct($client);
         $this->comments = new TicketComments($client);
+        $this->tags = new Tags($client);
```
Comment (jwswj) [inline src/Zendesk/API/Resources/Tickets.php]: I'm not a big fan of this. Any way we could define these somewhere else?

**Comment by jwswj** [comment]
This is looking good though

File: src/Zendesk/API/Resources/Tickets.php (hunk 3 above)
Comment (joseconsador) [inline src/Zendesk/API/Resources/Tickets.php]: This is solved via the PR https://github.com/zendesk/zendesk_api_client_php/pull/102

File: src/Zendesk/API/Resources/Tags.php (hunk 2 above)
Comment (joseconsador) [inline src/Zendesk/API/Resources/Tags.php]: Yes

File: src/Zendesk/API/Resources/Tags.php (hunk 1 above)
Comment (joseconsador) [inline src/Zendesk/API/Resources/Tags.php]: :haircut:

File: src/Zendesk/API/Resources/Tickets.php (hunk 3 above)
Comment (joseconsador) [inline src/Zendesk/API/Resources/Tickets.php]: :haircut:

File: src/Zendesk/API/Resources/Tags.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+/**
+ * The Tags class exposes methods as detailed on http://developer.zendesk.com/documentation/rest_api/tags.html
+ *
+ * @package Zendesk\API
+ */
+class Tags extends ResourceAbstract
+{
+
+    const OBJ_NAME = 'tags';
```
Comment (atroche) [inline src/Zendesk/API/Resources/Tags.php]: is this supposed to be `tag`?

**Comment by atroche** [comment]
:+1:

File: src/Zendesk/API/Resources/Tags.php (hunk 4 above)
Comment (joseconsador) [inline src/Zendesk/API/Resources/Tags.php]: There's no tag.json endpoint, so I'm guessing that's why it's always plural.

**Pull request merged**